tests/bsim/babblekit/include/babblekit/testcase.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_tracing.h"
+#include "bs_types.h"
+#include "bstests.h"
+
+extern enum bst_result_t bst_result;
+
+/*
+ * @brief Mark the test as in progress
+ *
+ * Call this at the start of the test entry point.
+ *
+ * @param ... format-string and arguments to print to console
+ *
+ */
+#define TEST_START(msg, ...)                                                                       \
+	do {                                                                                       \
+		bst_result = In_progress;                                                          \
+		bs_trace_info_time(2, "Test start: " msg "\n", ##__VA_ARGS__);                     \
```

Comment:
PSA: Explanation of `, ##__VA_ARGS__`: https://stackoverflow.com/questions/5588855/standard-alternative-to-gccs-va-args-trick